packages/nocodb/src/lib/db/sql-client/lib/oracle/OracleClient.ts

```diff
@@ -1580,6 +1589,63 @@ class OracleClient extends KnexClient {
     return result;
   }
 
+  // https://chartio.com/resources/tutorials/how-to-define-an-auto-increment-primary-key-in-oracle/
+  // fallback for older version to auto generate auto incremented pk using sequence and trigger
+  async afterTableCreate(args) {
+    const result = { upStatement: [], downStatement: [] };
+    let upQuery = '';
+    let downQuery = '';
+
+    const pk = args.columns.find((c) => c.pk);
+    if (!pk || !this.useTriggerBasedAI) return result;
```

Comment:
Just to confirm should we skip if no pk is present? Isn't it problematic for us to handle records on table without pk.